rir/src/interpreter/interp.c

```diff
@@ -333,21 +345,56 @@ static SEXP rirCallClosure(SEXP call, SEXP env, SEXP callee, SEXP actuals,
             optimizing = true;
 
             Function* oldFun = fun;
-            SEXP oldBody = body;
-            cp_pool_add(ctx, oldBody);
-            body = globalContext()->optimizer(callee);
+            SEXP oldFunStore = funStore;
+            cp_pool_add(ctx, oldFunStore);
+            funStore = globalContext()->optimizer(funStore);
 
             // TODO: there might be promises with references to the old code!
             // Therefore we keep it around.
             // TODO: first I tried to use R_PreserveObject, but it did not work
             // for large vectors, not sure why, need to investigate
-            cp_pool_add(ctx, body);
+            cp_pool_add(ctx, funStore);
```

Comment:
btw, this should no longer be necessary, since all versions are now reachable by the gc through the dispatch table

Reply:
OK, thanks.